plugins/org.eclipse.n4js/src/org/eclipse/n4js/validation/validators/N4JSTypeValidator.xtend

```diff
@@ -518,27 +520,54 @@ class N4JSTypeValidator extends AbstractN4JSDeclarativeValidator {
 				return;
 			}
 
+			val usedInSpecConstructor = objectLiteral.eContainer?.eContainer instanceof NewExpression;
```

Comment:
This will produce false positives (not every object literal passed into a constructor is the spec argument to a spec constructor).
The rule must be something like: (a) the object literal must be an argument to a ctor (b) the formal parameter of that ctor receiving the object literal as argument must be annotated with `@Spec`. (note that spec ctors may have more parameters than the spec parameter)

As Xpect test:
```javascript
class C {
	public prop: string;
	private propOther: string;

	constructor(arg0: ~i~this, @Spec arg1: ~i~this, arg2: ~i~this) {}
}

class D {
	constructor(arg0: ~C) {}
}

let c: ~C;

// works XPECT warnings --> "propUnknown is not defined in ~C; it will not be accessible from c." at "propUnknown"
c = {prop: "", propOther: "", propUnknown: ""};

function foo(arg: ~C) {}
// works XPECT warnings --> "propUnknown is not defined in ~C; it will not be accessible from the receiving parameter." at "propUnknown"
foo({prop: "", propOther: "", propUnknown: ""});

// this is not a spec constructor XPECT FIXME warnings --> "propUnknown is not defined in ~C; it will not be accessible from the receiving parameter." at "propUnknown"
new D({prop: "", propOther: "", propUnknown: ""});


new C(
	// not a spec argument XPECT FIXME warnings --> "propUnknown is not defined in ~C; it will not be accessible from the receiving parameter." at "propUnknown"
	{prop: "", propOther: "", propUnknown: ""},
	// works XPECT warnings ---
	// "propOther is not defined in ~i~this[C]; it will not have any effect in the spec constructor." at "propOther"
	// "propUnknown is not defined in ~i~this[C]; it will not have any effect in the spec constructor." at "propUnknown"
	// ---
	{prop: "", propOther: "", propUnknown: ""},
	// not a spec argument XPECT FIXME warnings --> "propUnknown is not defined in ~C; it will not be accessible from the receiving parameter." at "propUnknown"
	{prop: "", propOther: "", propUnknown: ""}
);
```